### PR TITLE
Refactors lecture-view.js to include classes

### PR DESCRIPTION
**Pull request by jaysaleh**
This pr refactors `lecture-view.js` to include classes.

### CODE REVIEW

File: src/main/webapp/view/video/video.js [hunk 1]
```diff
@@ -12,15 +12,26 @@
...
 import {secondsToMilliseconds} from '../../timestamps.js';
 
 const SCRIPT = 'script';
 
 /** Initializes and stores video player information. */
 export default class Video {
+  #lecture;
+  #synchronizer;
+
+  /** Creates a new `Synchronizer` linked to `this`. */
```
Comment (amuamushu) [inline src/main/webapp/view/video/video.js:25]: I find it weird the constructor only mentions creating a new synchronizer but not a new Video instance. I think `lecture` should be mentioned as well.

File: src/main/webapp/view/view.js [hunk 2]
```diff
@@ -12,66 +12,75 @@
...
+    return urlParams.get(PARAM_ID);
+  }
 
-/** Sets the lecture name in `header-text`. */
-function setLectureName() {
-  const headerText = document.getElementById('header-text');
-  headerText.innerText = window.LECTURE.lectureName;
+  /** Sets the lecture name in `header-text`. */
+  setLectureName() {
+    const headerText = document.getElementById('header-text');
```
Comment (amuamushu) [inline src/main/webapp/view/view.js:81]: Can you make `header-text` a constant?

File: src/main/webapp/view/view.js [hunk 3]
```diff
@@ -12,66 +12,75 @@
...
-/** Sets {@code window.LECTURE} as Lecture for view page. */
-getLecture().then((lecture) => {
-  window.LECTURE = lecture;
-  initialize();
-});
+    // TODO: Make these private once event controller is created.
+    View.video = new Video(this.#lecture);
+    // TODO: Move TranscriptArea initialization outside of initialize()
+    // and replace string parameter with a controller object.
+    View.transcript = new TranscriptArea(this.#lecture, 'event controller');
```
Comment (amuamushu) [inline src/main/webapp/view/view.js:47]: Just fyi, the `TranscriptArea` currently doesn't take in a lecture parameter. I added an issue to remind myself to add it later #299.

File: src/main/webapp/view/view.js [hunk 4]
```diff
@@ -12,66 +12,75 @@
...
 import TranscriptArea from './transcript/transcript-area.js';
 import Video from './video/video.js';
 
 const ENDPOINT_LECTURE = '/lecture';
 
 const PARAM_ID = 'id';
 
-// TODO: Remove global scope and add to view object.
-window.video = new Video();
+export default class View {
```
Comment (amuamushu) [inline src/main/webapp/view/view.js:23]: please add a class header

File: src/main/webapp/view/discussion/discussion.js [hunk 5]
```diff
@@ -12,26 +12,13 @@
...
-export async function intializeDiscussion() {
-  discussion = new DiscussionArea(window.LECTURE);
-  discussion.initialize();
-  // This is used as the `onclick` handler of the new comment area submit
-  // button. It must be set after discussion is initialized.
-  window.postNewComment = discussion.postNewComment.bind(discussion);
-}
-
 /** Seeks discussion to `timeMs`. */
 export function seekDiscussion(timeMs) {
```
Comment (n-wach) [inline src/main/webapp/view/discussion/discussion.js:22]: can we delete this?  or are we still waiting on event controller stuff

File: src/main/webapp/view/view.js [hunk 6]
```diff
@@ -12,66 +12,75 @@
...
+  #lecture;
+  #lectureId;
 
-/* exported LECTURE_ID */
-window.LECTURE_ID = getLectureId();
+  /** Sets `lectureId`, `lecture` and initializes view page components. */
+  constructor() {
+    this.#lectureId = this.getLectureId();
+    this.getLecture().then((lecture) => {
```
Comment (n-wach) [inline src/main/webapp/view/view.js:30]: probably shouldn't call this in the constructor...

File: src/main/webapp/view/view.js [hunk 7]
```diff
@@ -12,66 +12,75 @@
...
-  window.video.loadVideoApi();
-  intializeDiscussion();
-  // TODO: Move TranscriptArea initialization outside of initialize()
-  // and replace string parameter with a controller object.
-  const transcript = new TranscriptArea('event controller');
-  await transcript.loadTranscript();
-}
+    await View.video.loadVideoApi();
+    await View.transcript.loadTranscript();
+    await View.discussion.initialize();
```
Comment (n-wach) [inline src/main/webapp/view/view.js:52]: why are all of these static?

File: src/main/webapp/view/view.js (hunk 4 above)
Comment (n-wach) [inline src/main/webapp/view/view.js:23]: can we rename this `LectureView`? I think `View` is too generic

File: src/main/webapp/view/discussion/discussion.js (hunk 5 above)
Comment (n-wach) [reply src/main/webapp/view/discussion/discussion.js:22]: the `seekDiscussion` function

File: src/main/webapp/view/view.js (hunk 3 above)
Comment (jaysaleh) [reply src/main/webapp/view/view.js:47]: I addressed it here

File: src/main/webapp/view/discussion/discussion.js (hunk 5 above)
Comment (jaysaleh) [reply src/main/webapp/view/discussion/discussion.js:22]: waiting on the event controller stuff

File: src/main/webapp/view/view.js (hunk 7 above)
Comment (jaysaleh) [reply src/main/webapp/view/view.js:52]: They need to be referenced for the synchronization to work right now

File: src/main/webapp/view/view.js (hunk 4 above)
Comment (jaysaleh) [reply src/main/webapp/view/view.js:23]: renamed

File: src/main/webapp/view/view.js (hunk 3 above)
Comment (amuamushu) [reply src/main/webapp/view/view.js]: Gotcha, sorry must have missed it.

File: src/main/webapp/view/view.js (hunk 6 above)
Comment (jaysaleh) [reply src/main/webapp/view/view.js]: fixed

File: src/main/webapp/synchronizer.js [hunk 8]
```diff
@@ -29,13 +29,19 @@ const transcriptSeeker = new TranscriptSeeker('event controller');
  * time.
  */
 export default class Synchronizer {
+  #video;
+
+  constructor(video) {
+    this.#video = video;
+  }
```
Comment (billie-cat) [inline src/main/webapp/synchronizer.js:36]: Does this PR need to be synced? I saw this code in another PR.

Comment (jaysaleh) [reply src/main/webapp/synchronizer.js]: synced

**Pull request merged**